src/formats/yarle/utils/folder-utils.ts

```diff
@@ -118,8 +120,42 @@ export const clearResourceDir = (note: any): void => {
 	clearDistDir(resPath);
 	resourceDirClears.set(resPath, clears + 1);
 };
+export const getNotebookNameAndFolderNames = (basename: string): {notebookName: string, notebookFolderNames: string[]} => {
+	const notebookFolderNames = basename.split('@@@');
 
-export const setPaths = (enexSource: PickedFile): void => {
+	let notebookName = notebookFolderNames.pop();
+	if (!notebookName){
+		notebookName = basename;
+	}
+	return {
+		notebookName,
+		notebookFolderNames
+	}; 
+};
+
+
+export const getNotebookStackedEnex = (baseEnex: PickedFile ): NodePickedFile => {
+	if (!(baseEnex instanceof NodePickedFile)) throw new Error('Evernote import currently only works on desktop');
+
+	const { notebookName } =  getNotebookNameAndFolderNames(baseEnex.basename);
+
+	const stackedPickedFile = new NodePickedFile((baseEnex as NodePickedFile).filepath);
+	stackedPickedFile.fullpath = replaceLastOccurrenceInString(baseEnex.fullpath, baseEnex.basename, notebookName || baseEnex.basename);
```

Comment:
@lishid Ohh, okay, now I get it, thanks for the clarification! Changes made to make it work with untouchable NodePickedFiles. :)

Reply:
As I understand it, what's happening here is that we are finding some code down the line that uses these fields, and we'd like to change their behavior to use a different names/paths. Doing it this way is a "shortcut"; while it results in minimal amount of changes, it also makes the code much more confusing. Similar to what I said in the other comment, when you have a field that is defined to mean one thing, and you use it for a slightly different purpose (and thus modify the contents to fit your alternative needs), it creates a chance of confusion, or at the very least, forces the reader to "keep in mind" these extra concerns. This is often where bugs can slip in, especially if there are many different little things one needed to keep in mind when working with the code.

Usually the best way to do this is being more explicit with the code to say what you intend with the variables and fields. Passing additional vars with names that indicate what they do (or is intended for) helps a lot.

Reply:
Yup the new changes works for me.

Reply:
@lishid , sorry, I had a bad feeling about this, so I tried to improve the code further, introduced a couple of new functions to decrease the level of the possible confusion. Hope it looks better for you too.

Reply:
Thanks. I would say don't worry too much about it. Yarle is unfortunately written in a very different paradigm of programming than the rest of the importer. It uses global singleton objects and naked functions with context being passed sometimes, set in the global object at other times.

I decided not to invest too much effort to rewrite yarle as it would not be worth the time for a functionally equivalent package.

Reply:
okay then we can move forward with it, right?